lib/onelogin/ruby-saml/response.rb

```diff
@@ -29,19 +40,25 @@ def initialize(response, options = {})
         @document = XMLSecurity::SignedDocument.new(@response, @errors)
       end
 
+      # Validates the SAML Response with the default values (soft = true)
+      # @return [Boolean] TRUE if the SAML Response is valid
+      #
       def is_valid?
         validate
       end
 
-      def validate!
-        validate(false)
-      end
-
-      def errors
-        @errors
+      # Validates the SAML Response (soft = false)
+      # @param soft [Boolean] soft Enable or Disable the soft mode (In order to raise exceptions when the response is invalid or not)
+      # @param request_id [String|nil] request_id The ID of the AuthNRequest sent by this SP to the IdP (if was sent any)
+      # @return [Boolean] TRUE if the SAML Response is valid
+      # @raise [ValidationError] if soft == false and validation fails
+      #
+      def validate!(soft=false, request_id = nil)
```

Comment:
sometimes this kind of parameters makes it a lot harder to know what is going on exactly. Instead of method(true, 1234), use method(:soft => true, :request_id => 1234)